email_totals/handler.py

```diff
@@ -0,0 +1,315 @@
+            else:
+                raise ValueError(f"Too many keys: {group['Keys']}")
+    return all_costs
+
+
+def filter_email_list(all_totals, team_sage):
+    '''
+    Remove external and invalid emails from the dictionary
+     * Don't send emails for totals less than a specified minimum
+     * Don't send emails to synapse users that are not members of team Sage
+     * Optionally only send emails to an approved list of recipients (for testing)
+    '''
+
+    report_dict = {}
+
+    # get the minimum user total for sending an email
+    min_value = float(os.environ['MINIMUM'])
+
+    # get the list of approved recipients from an environment variable
+    restrict_to = os.environ['APPROVED'].split(',')
+
+    # if this environment variable is set, only send emails to recipients
+    # listed as approved recipients, otherwise send all valid emails
+    if os.environ['RESTRICT'] == 'True':
+        restrict_send = True
+        print(f'Only sending emails to: {restrict_to}')
+    else:
+        restrict_send = False
+
+    # loop over all our totals
+    for k, v in all_totals.items():
+        if v < min_value:
+            print(f"Skipping entry less than ${min_value}: {k} (${v})")
+
+        elif restrict_send:
+            if k in restrict_to:
+                report_dict[k] = v
+            else:
+                print(f"Restricting email: {k} (${v})")
+
+        else:
+            if k.endswith(sagebase_email) or k.endswith(sagebio_email):
+                report_dict[k] = v
+
+            elif k.endswith(synapse_email):
+                if k in team_sage:
+                    report_dict[k] = v
+                else:
+                    print(f"Skipping external synapse user: {k} (${v})")
+
+            else:
+                print(f"Skipping invalid email: {k} (${v})")
+
+    return report_dict
+
+
+#--------------------------------------------------------------------------------------------------
+# Compile HTML for E-mail Body
+
+def create_and_send_emails(target_totals, compare_totals):
+    '''
+    Build an HTML email from a string template and give it to our SES client
+    '''
+
+    html_title = '<h2>AWS Monthly Cost Report Summary</h2><br/>'
+
+    def _build_email(target_total, previous_total):
+        '''
+        The actual string template for the email
+        '''
+
+        if previous_total:
+            pct_change = (target_total / previous_total) - 1
+        else:
+            pct_change = 1
+
+        return ('You are receiving this email because you have been tagged as a '
+                f'resource owner in AWS. Your tagged resources total ${target_total:.2f} '
+                f'for the past month ({pct_change:.2%} change from the month prior). '
+                'You can use AWS Cost Explorer to analyze those expenses: '
+                '<a href="https://sagebionetworks.jira.com/wiki/spaces/IT/pages/2756935685/Using+AWS+Cost+Explorer">'
+                'Using AWS Cost Explorer</a>')
```

Comment:
don't we need to provide AWS account info to let users know which account(s) they need to log into to see the detailed info?

Reply:
no, by deploying this to the organizations account it aggregates data from all of the member accounts, and the documentation linked in the email recommends logging in to cost explorer in that account to track costs across all member accounts.